images/kindnetd/cmd/kindnetd/subnets.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"os"
+	"regexp"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/utils/net"
+)
+
+const (
+	kubeadmClusterCIDRRegex   = `\s+podSubnet: (.*)\n`
+	kubeproxyClusterCIDRRegex = `\s+clusterCIDR: (.*)\n`
+)
+
+type kubeSubnets struct {
+	regex     string
+	configmap string
+	namespace string
+}
+
+func newKubeSubnets(regex, configmap, namespace string) *kubeSubnets {
+	return &kubeSubnets{
+		regex:     regex,
+		configmap: configmap,
+		namespace: namespace,
+	}
+}
+
+func (ks *kubeSubnets) Get(clientset *kubernetes.Clientset) (string, error) {
+	cidrRegexp := regexp.MustCompile(ks.regex)
+	configMap, err := clientset.CoreV1().ConfigMaps(ks.namespace).Get(ks.configmap, metav1.GetOptions{})
+	if err != nil {
+		println("ERROR: " + err.Error())
+		return "", err
+	}
+	for _, data := range configMap.Data {
+		matches := cidrRegexp.FindStringSubmatch(data)
+		if len(matches) > 0 {
+			println("MATCH POD: " + matches[1])
+			return matches[1], nil
+		}
+	}
+	return "", nil
+}
+
+// getNoMasqueradeSubnets tries to get the POD networks subnets to not Masquerade them
+// It returns an array of strings with the Cluster CIDR subnets
+// It can only obtain the POD subnet parameter from one place for consistency
+// The order is:
+// 1. POD_SUBNET environment variables
+// 2. Pod subnet value in kubeadm configmap
+// 3. Cluster CIDR value in kube-proxy configmap
+func getNoMasqueradeSubnets(clientset *kubernetes.Clientset) ([]string, []string) {
```

Comment:
we run a goroutine with a kind of `IP Masquerade`agent in kindnet, 
https://kubernetes.io/docs/tasks/administer-cluster/ip-masq-agent/
so we don´t masquerade the traffic between pods, but it can receive more CIDR.
However, you are right, since this is only returning the cluster subnets